emcc.py

```diff
@@ -1990,8 +1991,18 @@ def compile_source_file(i, input_file):
     link_flags = [l for l in link_flags if l[1] not in ('-lm', '-lrt', '-ldl', '-lpthread')]
     temp_files = filter_out_dynamic_libs(temp_files)
 
-    # Decide what we will link
-    if not (shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES):
+    link_to_object = final_suffix not in executable_endings
+    using_lld = shared.Settings.WASM_BACKEND and not (link_to_object and not shared.Settings.WASM_OBJECT_FILES)
+    if using_lld:
+      # Filter out link flags that lld doesn't support.  bind_at_load is often
+      # passed on OSX because libtool/autoconf add this link flag.
+      def supported(f):
+        if any(f.startswith(p) for p in ('-bind_at_load','-rpath')):
```

Comment:
lists like this worry me - we're basically responding to bug reports and fixing things, but there may be a long tail of more flags that break?

Reply:
Right.. but ideally we would push back on such cases and say "hey .. you are using bad link flags.. you shouldn't expect that to work".   `-bind_at_load` is a really annoying case because it deep in the heart of libtool so its hard to ask users to strip it out.

In most cases when you have a link flag we don't support you should really remove it.

In other words, we can push back hard against growing this list. 

On the other hand, I do see that hisotically emscripten has tried to be all things to all people by silently ignore such invalid flags in order to enable things to get running quicker.   Perhaps this is another case for STRICT=1.. where in STRICT=0 mode (i.e. sloppy mode) we could silent ignore unknown link flags.

Reply:
Another reason I (selfishly) prefer a blacklist to whitelist here is that then I can play with new linker flags using `-Wl,--new-flag` without having emscripten strip this.

Reply:
But .. I guess as long there is warning.. I'll switch to whitelist if you prefer?

Reply:
Well, I honestly don't know what's best here... I am still somewhat worried, but I agree with the points you made, lgtm.

Reply:
I'll land this now since it fixes the bots.. then revisit/refactor.